apps/builder/app/builder/features/topbar/publish.tsx

```diff
@@ -292,14 +295,13 @@ const Publish = ({
 
       <Tooltip
         content={
-          publishState !== "idle" || isPublishing
-            ? "Publish process in progress"
-            : undefined
+          isPublishInProgress ? "Publish process in progress" : undefined
         }
       >
         <Button
           color="positive"
-          disabled={publishState !== "idle" || isPublishing}
+          disabled={isPublishInProgress}
+          prefix={isPublishInProgress && <LoadingAnimatedIcon />}
```

Comment:
as a workaround its fine, but we need this as a pending state on the button for everything to use it

Reply:
Until no design this should be prefered